pvfactors/engine.py

```diff
@@ -138,26 +136,20 @@ def run_timestep(self, idx):
         if self.skip_step[idx]:
             pvarray = None
         else:
-            # Update parameters
-            self.params.update(
-                {'solar_zenith': self.solar_zenith[idx],
-                 'solar_azimuth': self.solar_azimuth[idx],
-                 'surface_tilt': self.surface_tilt[idx],
-                 'surface_azimuth': self.surface_azimuth[idx]})
-
-            # Create pv array
-            pvarray = self.cls_pvarray.from_dict(
-                self.params, surface_params=self.irradiance.params)
-            pvarray.cast_shadows()
-            pvarray.cuts_for_pvrow_view()
-
-            # Prepare inputs
-            geom_dict = pvarray.dict_surfaces
+            # To be returned at the end
+            pvarray = self.pvarray
+
+            # Transform pvarray to time step
+            pvarray.transform(idx)
```

Comment:
now ``pvarray`` is also using the "fit()/transform()" paradigm, just like the irradiance models